### PR TITLE
#273 fixed sorting in autocomplete and autocomplete-feed

### LINKED ISSUE #273: Autocomplete: items get sorted incorrectly, when using inputMap.
Hexagon version: 1.4.2

If i use Autocomplete with inputMap:
```
  new hx.AutoComplete('.autocomplete-objects', itemsObjects, {
    inputMap: function(item) {
      return item.id;
    }
```
then the `ids` are sorted incorrectly: "51" appears before "1".
Though, when using plain array without inputMap, sort is working properly.

See the plunker, demonstrating the bug:
https://plnkr.co/edit/sfg0qRk8lg49u32ZPCrJ?p=preview

Also here is a screenshot, what I see in this plunker:
hexagon-autocomplete-sort-bug

### PR DESCRIPTION
**Pull request by jonthans**
resolves #273

### CODE REVIEW

File: modules/autocomplete/main/index.coffee [hunk 1]
```diff
@@ -63,15 +63,9 @@ findTerm = (term, forceMatch) ->
             data = data.sort hx.sort.compare
         data
 
+    groupedActive = new hx.Map hx.groupBy remainingResults, (i) -> not i.disabled
 
-    remainingResults = heading.concat(
-      remainingResults.sort (a, b) ->
-        if not a.disabled and b.disabled then -1
-        else if a.disabled and not b.disabled then 1
-        else hx.sort.compare(a.full, b.full)
```
Comment (jonthans) [inline modules/autocomplete/main/index.coffee]: No need to sort here. Also, what was `.full`?

Comment (c-frater) [inline modules/autocomplete/main/index.coffee]: I think `.full` was something used in preferences for selecting locales - it should never have been in this file...

File: modules/autocomplete-feed/main/index.coffee [hunk 2]
```diff
@@ -33,11 +30,10 @@ class AutocompleteFeed
...
     # defined here so we can use the resolved options
     resolvedOptions.filter ?= (items, term) ->
-      hx.filter[resolvedOptions.matchType](items, term, resolvedOptions.filterOptions)
-        .sort (a, b) ->
-          if not a.disabled and b.disabled then -1
-          else if a.disabled and not b.disabled then 1
-          else 0
+      filtered = hx.filter[resolvedOptions.matchType](items, term, resolvedOptions.filterOptions)
+
+      groupedActive = new hx.Map hx.groupBy filtered, (i) -> not i.disabled
```
Comment (jmsmyth) [inline modules/autocomplete-feed/main/index.coffee]: Some brackets would be useful here. Its not clear what belongs to what..

File: modules/autocomplete-feed/main/index.coffee [hunk 3]
```diff
@@ -79,10 +75,13 @@ class AutocompleteFeed
       filterAndCallback = (unfilteredItems) ->
         filteredItems = _.options.filter(unfilteredItems, term)
         if _.options.showOtherResults
-          otherResults = unfilteredItems.filter (datum) ->
+          unpartitioned = unfilteredItems.filter (datum) ->
               filteredItems.indexOf(datum) is -1
             .sort sortItems(_.options.valueLookup)
 
+          groupedActive = new hx.Map hx.groupBy unpartitioned, (i) -> not i.disabled
```
Comment (jmsmyth) [inline modules/autocomplete-feed/main/index.coffee]: Brackets again

File: modules/autocomplete/main/index.coffee [hunk 4]
```diff
@@ -222,12 +216,9 @@ class AutoComplete extends hx.EventEmitter
...
 
       @options.filter ?= (arr, term) ->
-        hx.filter[self.options.matchType](arr, term, self.options.filterOptions)
-          .sort (a, b) ->
-            if not a.disabled and b.disabled then -1
-            else if a.disabled and not b.disabled then 1
-            else hx.sort.compare(a, b)
-
+        filtered = hx.filter[self.options.matchType](arr, term, self.options.filterOptions)
+        groupedActive = new hx.Map hx.groupBy filtered, (i) -> not i.disabled
```
Comment (jmsmyth) [inline modules/autocomplete/main/index.coffee]: ^^ :-p

File: modules/autocomplete/test/spec.coffee [hunk 5]
```diff
@@ -4,3 +4,21 @@ describe 'autocomplete', ->
...
+  it 'should sort items correctly when using objects', ->
+    itemsObjects = hx.range(100).map (i) -> id: i
+
+    ac = new hx.AutoComplete hx.detached('div').node(), itemsObjects
+
+    ac.show()
+    ac._.menu.items().map(({id}) -> id).should.eql(hx.range(100))
+
+  it 'should sort disabled items correctly', ->
+    itemsObjects = hx.range(100).map (i) -> id: i, disabled: i % 2 == 0
```
Comment (jmsmyth) [inline modules/autocomplete/test/spec.coffee]: Another coffeescript syntax ambiguity.. Is this being mapped to an object? Can it be made more explicit with curly brackets?

**Pull request merged**